cmd/system-probe/modules/network_tracer.go

```diff
@@ -77,15 +76,20 @@ var NetworkTracer = module.Factory{
 		nt := networkTracer{
 			tracer: t,
 		}
+		nt.done = done
 
-		go func() {
-			err := startRPCServer(nt)
-			if err != nil {
-				log.Errorf("Failed to start gRPC server: %v", err)
+		if ncfg.UseGRPC {
+			if ncfg.GRPCSocketFilePath == "" {
+				log.Errorf("unable to create a grpc server without a path for the socket file")
```

Comment:
trying to make a simpler version, let me know what do you think on it
```suggestion
				log.Error("grpc unix socket path is empty")
```